### DISCUSSION TITLE
one more question about ;dtmfmode=auto account parameter

### DISCUSSION THREAD

**Discussion by juha-h** [baresip/baresip#2983] [Q&A]
Wiki https://github.com/baresip/baresip/wiki/Accounts#dtmfmode tells about `;dtmfmode=auto` value
```
auto: It depends on the remote SDP which method is used. If there is an SDP format with name "telephone-event" then inbound DTMF is used, otherwise SIP INFO.
```
In the latter case, what happens if peer does not list INFO in its Allow header?  Is it so that then baresip does not send DTMF keys?

**Comment by juha-h** [comment]
If remote party of the call  does not advertise telephone-event nor INFO method, can baresip local party of the call somehow learn that the remote party is not able to receive DTMF?

**Reply by cspiel1** [reply]
Currently, baresip does not consider this option. In this case I am pretty sure, that it will ignore the Allow header and send a SIP INFO.

**Comment by juha-h** [comment]
I didn't find in `struct call` a field that would contain a list of Allow header tags.  When call is established and peer has included Allow header in INVITE or 200 OK, then such a list could be constructed.  Once done, ua would know if sending INFO makes sense.  INFO can be sent if allow list is empty or contains INFO tag.
